core/src/impl/Kokkos_ClockTic.hpp

```diff
@@ -27,8 +27,9 @@
 // To use OpenCL(TM) built-in intrinsics inside kernels, we have to
 // forward-declare their prototype, also see
 // https://github.com/intel/pti-gpu/blob/master/chapters/binary_instrumentation/OpenCLBuiltIn.md
-#if defined(KOKKOS_ENABLE_SYCL) && defined(KOKKOS_ARCH_INTEL_GPU) && \
-    defined(__SYCL_DEVICE_ONLY__)
+#if defined(KOKKOS_ENABLE_SYCL) &&                              \
+    defined(KOKKOS_ENABLE_IMPL_SYCL_RELOCATABLE_DEVICE_CODE) && \
+    defined(KOKKOS_ARCH_INTEL_GPU) && defined(__SYCL_DEVICE_ONLY__)
```

Comment:
So unless the user enables RDC, they now get a garbage clock tic on the device side.  Do I understand it right?
You might want to add a `FIXME_SYCL` here.

Reply:
Added one below where it's actually invoked.